src/main/java/org/springframework/integration/kafka/inbound/KafkaInboundGateway.java

```diff
@@ -212,6 +233,13 @@ protected AttributeAccessor getErrorMessageAttributes(Message<?> message) {
 			super(null, null);
 		}
 
+		@Override
+		public void onPartitionsAssigned(Map<TopicPartition, Long> assignments, ConsumerSeekCallback callback) {
+			if (KafkaInboundGateway.this.onPartitionsAssignedSeekCallback != null) {
+				KafkaInboundGateway.this.onPartitionsAssignedSeekCallback.accept(assignments, callback);
+			}
+		}
+
```

Comment:
Why only from the record listener?

Reply:
Huh? There is no `IntegrationBatchMessageListener` in the gateway.

Sorry, I'm not sure how to proceed.

Reply:
On other thought the batch processing does not make sense in the request-reply scenarios.
So, that's fully fine that we don't support `batch` there.

Please, elaborate with your concern.

Thanks